integrations/kube-agent-updater/pkg/maintenance/basichttp.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package maintenance
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/exp/slices"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/cache"
+)
+
+const (
+	maintenancePath = "emergency"
+	httpTimeout     = 10 * time.Second
+	cacheDuration   = time.Minute
+)
+
+var (
+	truthy = []string{"true", "True", "yes", "Yes"}
+	falsy  = []string{"false", "False", "no", "No"}
+)
+
+// basicHTTPMaintenanceClient retrieves the version from an HTTP endpoint
```

Comment:
It doesn't look like this client fetches a version - it looks like it actually fetches a boolean value.